.changeset/tidy-pugs-repeat.md

```
@@ -0,0 +1,7 @@
+---
+'@mastra/server': patch
+---
+
+Resolve the public origin from `X-Mastra-Public-Host` when present, ahead of `X-Forwarded-Host`.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)